.github/workflows/pypi-cd.yml

```diff
@@ -198,31 +198,64 @@ jobs:
                   path: python/wheels
                   if-no-files-found: error
 
-    publish-to-pypi:
+    build-source-dist-and-publish-to-pypi:
```

Comment:
I propose to create another job to build sources and add it there as a dependency (`needs` section)

Reply:
It will require to upload the build files and then download them, I think it's unnecessary and makes sense to keep it in the same job due to this reason